### PR TITLE
Removed old scheduler constructor.

### PR DESCRIPTION
**Pull request by k82cn**
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes # N/A

**Release note**:

```release-note-none
```

### CODE REVIEW

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -121,16 +121,6 @@ type Config struct {
 	StopEverything chan struct{}
 }
 
-// New returns a new scheduler.
-// TODO replace this with NewFromConfigurator.
-func New(c *Config) *Scheduler {
-	s := &Scheduler{
-		config: c,
-	}
-	metrics.Register()
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/scheduler.go]: What is registering metrics now?

Comment (k82cn) [inline plugin/pkg/scheduler/scheduler.go]: `New` is replaced by `NewFromConfigurator` now. `New` is not use in kube-scheduler but tests, I'm updating code for them.

File: plugin/pkg/scheduler/testutil.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	corelisters "k8s.io/kubernetes/pkg/client/listers/core/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/util"
+)
+
+type FakeConfigurator struct {
+	Config                         *Config
+	PriorityConfigs                []algorithm.PriorityConfig
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/testutil.go]: All of these seems to be unused in all usecases.
Unless you see a good reason to have them (I don't), I really suggest removing all of these fields.

And all accessort functions can simple return nils (which they already do) or errors "not implemented" (if it won't break anything).

Comment (k82cn) [inline plugin/pkg/scheduler/testutil.go]: Good point, I'll update it.

**Pull request merged**